lib/tx_build/manage_sell_offer.ex

```diff
@@ -0,0 +1,87 @@
+defmodule Stellar.TxBuild.ManageSellOffer do
+  @moduledoc """
+  Creates, updates, or deletes an offer.
+  """
+  import Stellar.TxBuild.OpValidate
+
+  alias Stellar.TxBuild.{Amount, Asset, OptionalAccount, Price}
+  alias StellarBase.XDR.{OperationBody, OperationType, Operations.ManageSellOffer, Int64}
+
+  @behaviour Stellar.TxBuild.XDR
+
+  @type t :: %__MODULE__{
+          selling: Asset.t(),
+          buying: Asset.t(),
+          amount: Amount.t(),
+          price: Price.t(),
+          offer_id: integer(),
+          source_account: OptionalAccount.t()
+        }
+
+  defstruct [
+    :selling,
+    :buying,
+    :amount,
+    :path,
+    :price,
+    :offer_id,
+    :source_account
+  ]
+
+  @impl true
+  def new(args, opts \\ [])
+
+  def new(args, _opts) when is_list(args) do
+    selling = Keyword.get(args, :selling)
+    buying = Keyword.get(args, :buying)
+    amount = Keyword.get(args, :amount)
+    price = Keyword.get(args, :price)
+    offer_id = Keyword.get(args, :offer_id)
+    source_account = Keyword.get(args, :source_account)
```

Comment:
Why not a pattern matching in the function declaration?

```elixir
def new([selling: selling, buying: buying, .....], _opts)
```

You won't even need the guard this way.

Reply:
Because all the operations have optional parameters, pattern matching will require several function clauses and code won't be easy to understand.

I've tried other options but IMO, this is the simplest and understandable way to collect data and then validate it.